File: katas/7 kyu/two-to-one/solution.js
// two-to-one
/* ------------------------------- MY SOLUTION ------------------------------ */
function longest(s1, s2) {
  s = (s1 + s2).split("")
  return s.filter((v, i, a) => a.indexOf(v) === i).sort().join("")
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
const longest = (s1, s2) => [...new Set(s1+s2)].sort().join('')
/* ---------------------------------- TEST ---------------------------------- */
const chai = require("chai");
const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
chai.config.truncateThreshold=0;

describe("longest",function() {
  it("Basic tests",function() {
      Test(longest("aretheyhere", "yestheyarehere"), "aehrsty")
      Test(longest("loopingisfunbutdangerous", "lessdangerousthancoding"), "abcdefghilnoprstu")
      Test(longest("inmanylanguages", "theresapairoffunctions"), "acefghilmnoprstuy")
  })})

File: katas/7 kyu/two-to-one/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def longest(s1, s2):
    alphabet = "abcdefghijklmnopqrstuvwxyz"
    s = s1 + s2
    y = ""
    for x in alphabet:
      if x not in s:
        continue
      if x in s:
        y = y + x
    return y
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def longest(a1, a2):
    return ''.join(sorted(set(a1 + a2)))
# ----------------------------------- TEST ----------------------------------- #
@test.describe("longest")
def tests():
    @test.it("basic tests")
    def basics():
        test.assert_equals(longest("aretheyhere", "yestheyarehere"), "aehrsty")
        test.assert_equals(longest("loopingisfunbutdangerous", "lessdangerousthancoding"), "abcdefghilnoprstu")
        test.assert_equals(longest("inmanylanguages", "theresapairoffunctions"), "acefghilmnoprstuy")
